### ISSUE TITLE
kotlin 1.4 suggenst sam conversion

### ISSUE THREAD

**Issue by zmunm** [detekt/detekt#3529]
## Expected Behavior of the rule

Similar to #3528, but slightly different
I would like to suggest that interfaces containing only a single abstract method should be prefixed with fun.

interfaces containing only a single abstract method
```kotlin
interface Foo {
    fun bar()
}
```

What I want
```
fun interface Foo {
    fun bar()
}
```

## Context

https://blog.jetbrains.com/kotlin/2019/12/what-to-expect-in-kotlin-1-4-and-beyond/#sam-conversions

**Comment by BraisGabin** [comment]
🤔 I do like the `fun interface`s but do it have sense that detekt promote them as a "standard". This is that kind of rule that I really want for my code base but it feels that detekt should not push the users to use `fun interface`s as a "good practice". And I know that a lot of people follow the detekt rules blindly so I think that we should just add rules that are real "good practices". Or create the ruleset "controversial" that we talked in other issues.

**Comment by zmunm** [comment]
I agree that this case feels `standard` by asking users to put `fun` before actually using it. and it can be false positive for interfaces that are always implemented 

What if I change the detection method? In my opinion, if that notify a place where actually used as an anonymous object, like #3528, doesn't seem to hurt the nature of the offer.

I attach a new case to compensate for my lack of English.

compliant
```kotlin
interface Foo {
    fun bar()
}
```


non compliant
```kotlin
interface Foo {
    fun bar()
}

val listener = object : Foo {
    override fun bar() {
         ...
    }
}
```

**Comment by cortinico** [comment]
> What if I change the detection method? In my opinion, if that notify a place where actually used as an anonymous object, like #3528, doesn't seem to hurt the nature of the offer.

This would be interesting to try. I'm just unsure if it's feasible to develop such a rule as you would need to investigate both the declaration site and the implementation site.

I'm unsure if we have an easy way going from an `interface` declaration to check all the declarations to find for anonymous ones.

**Comment by schalkms** [comment]
@BraisGabin @cortinico this issue is referenced by multiple PRs. Isn't it obsolete?

**Comment by atulgpt** [comment]
@schalkms I don't think so as this issue talks about converting `interface` to `fun interface` while referencing PRs works as usage sites to give warning when some code which can leverage lambda syntax is not using so

**Comment by schalkms** [comment]
Then we should change the title, as it's misleading for fast readers.
